pkg/config/metadata/user_info_test.go

```diff
@@ -64,33 +80,28 @@ func TestUserInfoCall(t *testing.T) {
 func TestUserInfoCanceledContext(t *testing.T) {
 	ctrl := NewController(t)
 	defer ctrl.Finish()
+	ta := newUserInfoTestData(ctrl)
 
-	authContextMock := NewMockAuthContext(ctrl)
-	idConfEval := NewMockIdentityConfigEvaluator(ctrl)
-	oidcEvaluator, _ := identity.NewOIDC(fmt.Sprintf("http://%s", authServerHost), &authCredMock{})
-	idConfEval.EXPECT().GetOIDC().Return(oidcEvaluator)
-	authContextMock.EXPECT().GetHttp().Return(nil)
-	authContextMock.EXPECT().GetResolvedIdentity().Return(idConfEval, nil)
+	ta.authCredMock.EXPECT().GetCredentialsFromReq(Any()).Return("", nil)
+	ta.idConfEvalMock.EXPECT().GetOIDC().Return(ta.newOIDC)
+	ta.authContextMock.EXPECT().GetHttp().Return(nil)
+	ta.authContextMock.EXPECT().GetResolvedIdentity().Return(ta.idConfEvalMock, nil)
+
+	ta.cancel()
+	_, err := ta.userInfo.Call(ta.authContextMock, ta.ctx)
 
-	ctx, cancel := context.WithCancel(context.TODO())
-	cancel()
-	userInfo := UserInfo{oidcEvaluator}
-	_, err := userInfo.Call(authContextMock, ctx)
 	assert.Error(t, err, "context canceled")
 }
 
 func TestUserInfoMissingOIDCConfig(t *testing.T) {
 	ctrl := NewController(t)
 	defer ctrl.Finish()
+	ta := newUserInfoTestData(ctrl)
 
-	authContextMock := NewMockAuthContext(ctrl)
-	idConfEval := NewMockIdentityConfigEvaluator(ctrl)
-	oidcEvaluator, _ := identity.NewOIDC(fmt.Sprintf("http://%s", authServerHost), &authCredMock{})
-	otherOidcEvaluator, _ := identity.NewOIDC("http://wrongServer", &authCredMock{})
-	idConfEval.EXPECT().GetOIDC().Return(otherOidcEvaluator)
-	authContextMock.EXPECT().GetResolvedIdentity().Return(idConfEval, nil)
+	otherOidcEvaluator, _ := identity.NewOIDC(fmt.Sprintf("http://wrongServer"), ta.authCredMock)
```

Comment:
We should not use `fmt.Sprintf` without wildcard arguments. It was raising a warning before and I fixed it.

```suggestion
	otherOidcEvaluator, _ := identity.NewOIDC("http://wrongServer", ta.authCredMock)
```